File: tests/fixtures/dart/sample.dart
import 'dart:math';

/// 形状基类（Dart 用抽象类 + mixin 表达）。
abstract class Shape {
  /// 名字
  final String name;

  const Shape(this.name);

  double area();
}

mixin Walkable {
  String walk() => 'walk';
}

class Circle extends Shape with Walkable {
  final double radius;

  Circle(this.radius) : super('circle');

  @override
  double area() => radius > 0 ? pi * radius * radius : 0;
}

enum Kind { red, green }

int helper(int a) {
  for (var i = 0; i < a; i++) {
    if (i > 2 && i < 9) return i;
  }
  return -a;
}

File: tests/fixtures/cpp/sample.cpp
#include <string>
#include <vector>

/// 形状基类（C++ 用继承）。
class Shape {
public:
    explicit Shape(std::string name) : name_(std::move(name)) {}

    virtual double area() const = 0;

protected:
    std::string name_;
};

class Circle : public Shape {
public:
    Circle(double r) : Shape("circle"), r_(r) {}

    double area() const override {
        return 3.14 * r_ * r_;
    }

private:
    double r_;
};

struct Point {
    int x;
    int y;
};

enum class Kind { Red, Green };

double helper(double a) {
    return a > 0 ? a : -a;
}
